src/lib.rs

```diff
@@ -40,3 +40,12 @@ pub mod cpu;
 pub mod instruction;
 pub mod memory;
 pub mod registers;
+
+pub trait Variant {
```

Comment:
Some doc comments would help here.

Reply:
Just a basic description that there are different variants of the CPU, and that they handle operations differently.